### PR TITLE
AC_Fence: ensure fencepoint protocol not compiled in if not HAL_GCS_ENABLED

### PR DESCRIPTION
**Pull request by peterbarker**
also use GCS_SEND_TEXT a bit

### CODE REVIEW

**Comment by rmackay9** [comment]
To be fair we should consider pulling in commits from this pr https://github.com/ArduPilot/ardupilot/pull/25262

File: libraries/AC_Fence/AC_Fence_config.h [hunk 1]
```diff
@@ -11,5 +13,5 @@
 #endif
 
 #ifndef AC_POLYFENCE_FENCE_POINT_PROTOCOL_SUPPORT
-#define AC_POLYFENCE_FENCE_POINT_PROTOCOL_SUPPORT AP_FENCE_ENABLED
+#define AC_POLYFENCE_FENCE_POINT_PROTOCOL_SUPPORT HAL_GCS_ENABLED && AP_FENCE_ENABLED
```
Comment (amilcarlucas) [inline libraries/AC_Fence/AC_Fence_config.h:16]: Won't the GCS_SEND_TEXT macros take care of the case when HAL_GCS_ENABLED is zero?

Is this change really needed?

Comment (peterbarker) [reply libraries/AC_Fence/AC_Fence_config.h:16]: There are other bits in there that aren't gcs().send_text.

Also, the fencepoint protocol *is* mavlink, so it makes sense to gate it.

Comment (amilcarlucas) [reply libraries/AC_Fence/AC_Fence_config.h:16]: I need coffee, sorry. I read Fence instead of Fencepoint. :) :)

**Comment by peterbarker** [comment]
> To be fair we should consider pulling in commits from this pr #25262

I've rebased on top of a cherry-picked patch from that PR and force-pushed this branch

**Review by tridge — APPROVED** [review]
